File: workout_logger_app/lib/models/routine.dart
class Routine {
  final String id;
  final String name;
  final DateTime createdAt;

  Routine({required this.id, required this.name, required this.createdAt});

  factory Routine.fromJson(Map<String, dynamic> json) {
    return Routine(
      id: json['_id'],
      name: json['name'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}

File: workout-logger-backend/models/Routine.js
const mongoose = require('mongoose');

const RoutineSchema = new mongoose.Schema({
  userId: { type: String, required: true },
  name: { type: String, required: true },
  createdAt: { type: Date, default: Date.now }
});

RoutineSchema.index({ userId: 1, name: 1 });

module.exports = mongoose.model('Routine', RoutineSchema);
